<launch>

<arg name="robotname" default="diago_0" />
<arg name="plan_folder" value="$(find robocupathome_pnp)/plans" />
<arg name="PNPGUI" default="false" />

<group ns="$(arg robotname)">

    <param name="robotname" value="$(arg robotname)" />
    <param name="robot_name" value="$(arg robotname)" />

    <!-- PNP engine -->

    <node pkg="pnp_ros" type="pnp_node" name="pnp" args="" output="screen">
        <param name="plan_folder" value="$(arg plan_folder)" />
        <param name="current_plan" value="stop" />
        <param name="use_java_connection" value="$(arg PNPGUI)" />
    </node>

    <!-- PNP action server -->

    <node pkg="robocupathome_pnp" type="rcathome_pnpas" name="robocupathome_pnpas" args="" output="screen">      
    </node>


    <!-- PNP ROSPlan -->

    <node pkg="pnp_rosplan" type="pnp_rosplan_node" name="pnp_rosplan" args="" output="screen">
        <param name="plan_folder" value="$(arg plan_folder)" />
    </node>

    <!-- CONDITIONS -->

    <!-- provide data for persondetected and personhere conditions -->
    <node pkg="laser_analysis" name="laserobstacle" type="laserobstacle" args=""  output="screen">
	    <param name="map_frame_id" value="/$(arg robotname)/map" />
	    <param name="laser_frame_id" value="/$(arg robotname)/laser_frame" />
	    <param name="map_distance" value="0.5" />
        <param name="obstacle_margin_x_min" value="0" />
        <param name="obstacle_margin_x_max" value="6" />
        <param name="obstacle_margin_y_min" value="-1" />
        <param name="obstacle_margin_y_max" value="1" />
    </node>

    <!-- ACTIONS -->

    <node pkg="rococo_navigation" name="turn" type="turn_node" args="-server $(arg robotname)"  output="screen">
	</node>

    <node pkg="rococo_navigation" name="followperson" type="follow_person_node" args="-server $(arg robotname)"  output="screen">
	</node>

    <!-- AUX -->
    
    <node pkg="tcp_interface" type="tcp_interface" name="tcp_interface" args="" output="screen">
        <param name="TCP_server_port" value="9000"/>
        <param name="UDP_server_port" value="9100"/>
        <param name="network_interface" value="wlan0"/>
        <param name="UDP_team_min_port" value="9100"/>
        <param name="UDP_team_max_port" value="9110"/>
        <param name="config_file" value="$(find tcp_interface)/config/UDPpeers.cfg"/>    
    </node>


</group><!-- ns -->

</launch>

